qiskit/converters/dag_to_circuit.py

```diff
@@ -23,24 +25,35 @@ def dag_to_circuit(dag):
     Return:
         QuantumCircuit: the circuit representing the input dag.
     """
-    circuit = QuantumCircuit()
-    random_name = QuantumCircuit.cls_prefix() + \
-        str(''.join(random.choice(string.ascii_lowercase) for i in range(8)))
-    circuit.name = dag.name or random_name
+    qregs = collections.OrderedDict()
     for qreg in dag.qregs.values():
-        circuit.add_register(qreg)
+        qreg_tmp = QuantumRegister(qreg.size, name=qreg.name)
+        qregs[qreg.name] = qreg_tmp
+    cregs = collections.OrderedDict()
     for creg in dag.cregs.values():
-        circuit.add_register(creg)
+        creg_tmp = ClassicalRegister(creg.size, name=creg.name)
+        cregs[creg.name] = creg_tmp
+
+    random_name = QuantumCircuit.cls_prefix() + \
+        str(''.join(random.choice(string.ascii_lowercase) for i in range(8)))
+    name = dag.name or random_name
```

Comment:
Probably a question for the previous implementation, but would it be more consistent to allow `name=None`, and delegate the picking of a name to `QuantumCircuit.__init__`? Mostly for narrowing down the amount of places where names are auto-assigned and potentially help with reproducibility.

Reply:
Sure, I can easily remove this I just swap out random_name for None and then it'll just use the default behavior for QuantumCircuit(). But I'll defer to @ajavadia since there may be a reason this function was setting a random_name here.

Reply:
Yeah if you can ensure that QuantumCircuit will assign a name, I'm fine. I remember there were some tests that failed with QobjValidationError, because the circuit name was None and that's not valid under the schema. That's why I just made sure to explicitly assign a name. But admittedly didn't think much about it.

Reply:
I've removed random_name in the latest revision. I didn't see any test failures locally because of it.